src/shims/time.h

```diff
@@ -152,9 +152,9 @@ _dispatch_approximate_time(void)
 	struct timespec ts;
 	dispatch_assume_zero(clock_gettime(CLOCK_UPTIME_FAST, &ts));
 	return _dispatch_timespec_to_nano(ts);
-#elif defined(__linux__)
+#elif HAVE_DECL_CLOCK_MONOTONIC_COARSE
```

Comment:
please keep the `if defined(__linux__)` because CLOCK_MONOTONIC is supposed to tick during machine sleeps if you look at POSIX, and it happens to not be the case on linux which eventually became ABI on this platform.

However if a platform has (1) MONOTONIC_COARSE and not UPTIME_FAST, and (2) is POSIX compliant, then this would cause a bug for the said platform.

However you're right that MONOTONIC_COARSE is what we want here.